File: core/src/nl/t64/game/rpg/subjects/ProfileSubject.java
package nl.t64.game.rpg.subjects;

import nl.t64.game.rpg.ProfileManager;

import java.util.ArrayList;
import java.util.List;


public class ProfileSubject {

    private final List<ProfileObserver> observers = new ArrayList<>();

    public void addObserver(ProfileObserver observer) {
        observers.add(observer);
    }

    public void removeObserver(ProfileObserver observer) {
        observers.remove(observer);
    }

    public void notifyCreateProfile(ProfileManager profileManager) {
        observers.forEach(observer -> observer.onNotifyCreateProfile(profileManager));
    }

    public void notifySaveProfile(ProfileManager profileManager) {
        observers.forEach(observer -> observer.onNotifySaveProfile(profileManager));
    }

    public void notifyLoadProfile(ProfileManager profileManager) {
        observers.forEach(observer -> observer.onNotifyLoadProfile(profileManager));
    }

}

File: core/src/nl/t64/game/rpg/subjects/ProfileSubject.kt
package nl.t64.game.rpg.subjects

import nl.t64.game.rpg.ProfileManager


class ProfileSubject {

    private val observers: MutableList<ProfileObserver> = ArrayList()

    fun addObserver(observer: ProfileObserver) {
        observers.add(observer)
    }

    fun removeObserver(observer: ProfileObserver) {
        observers.remove(observer)
    }

    fun notifyCreateProfile(profileManager: ProfileManager) {
        observers.forEach { it.onNotifyCreateProfile(profileManager) }
    }

    fun notifySaveProfile(profileManager: ProfileManager) {
        observers.forEach { it.onNotifySaveProfile(profileManager) }
    }

    fun notifyLoadProfile(profileManager: ProfileManager) {
        observers.forEach { it.onNotifyLoadProfile(profileManager) }
    }

}
